adminSiteClient/FilesIndexPage.tsx

```diff
@@ -0,0 +1,373 @@
+            </div>
+            <Tooltip title="Copy file URL">
+                <button
+                    className="file-viewer__copy-button"
+                    onClick={() => copyToClipboard(fileUrl)}
+                >
+                    <FontAwesomeIcon icon={faCopy} />
+                </button>
+            </Tooltip>
+        </div>
+    )
+}
+
+function FolderButton({
+    name,
+    onClick,
+}: {
+    name: string
+    onClick: () => void
+}) {
+    return (
+        <button className="file-viewer__folder" onClick={onClick}>
+            {name}
+        </button>
+    )
+}
+
+function FileMapViewer({
+    fileMap,
+    currentPath,
+}: {
+    fileMap: FileMap
+    currentPath: string
+}) {
+    const history = useHistory()
+
+    const handleFolderClick = (folderName: string) => {
+        history.push(`?path=${urlJoin(currentPath, folderName)}`)
+    }
+
+    const pathSegments = currentPath.split("/").filter(Boolean)
+    const currentNode = R.pathOr(fileMap, pathSegments as any, fileMap)
+
+    return (
+        <div>
+            {Object.entries(currentNode).map(([key, value]) => {
+                if (checkIsFile(value)) {
+                    return <FileButton key={key} file={value} />
+                } else {
+                    return (
+                        <FolderButton
+                            key={key}
+                            name={key}
+                            onClick={() => handleFolderClick(key)}
+                        />
+                    )
+                }
+            })}
+        </div>
+    )
+}
+
+function SearchResultsViewer({
+    fileMap,
+    searchValue,
+}: {
+    fileMap: FileMap
+    searchValue: string
+}) {
+    const allFiles = useMemo(() => {
+        const flattenFiles = (
+            map: FileMap,
+            currentPath: string = ""
+        ): DbPlainFile[] => {
+            const files: DbPlainFile[] = []
+
+            Object.entries(map).forEach(([key, value]) => {
+                const fullPath = currentPath ? `${currentPath}/${key}` : key
+
+                if (checkIsFile(value)) {
+                    files.push(value)
+                } else {
+                    files.push(...flattenFiles(value, fullPath))
+                }
+            })
+
+            return files
+        }
+
+        return flattenFiles(fileMap)
+    }, [fileMap])
+
+    const filteredFiles = useMemo(() => {
+        return allFiles.filter(
+            (file) =>
+                file.filename
+                    .toLowerCase()
+                    .includes(searchValue.toLowerCase()) ||
+                file.path.toLowerCase().includes(searchValue.toLowerCase())
+        )
+    }, [allFiles, searchValue])
+
+    // Show search view immediately when user starts typing
+    if (!searchValue.trim()) {
+        return (
+            <div style={{ color: "#666", fontStyle: "italic" }}>
+                Start typing to search files...
+            </div>
+        )
+    }
```

Comment:
if this is only shown when typing more than 4 (incl.) spaces (given the condition to show SearchResultsViewer in the first place), it doesn't seem very useful. We probably want to trim before checking for the length - and then swap the SearchResultsViewer for the FileMapViewer.